tests/security/static/scripts/init.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+REPO_ROOT=$(git rev-parse --show-toplevel)
+BUILD_PATH="$REPO_ROOT/build"
+STATIC_PATH="$REPO_ROOT/tests/security/static"
+CODEQL_CLI_VERSION=$(<"$STATIC_PATH/scripts/codeql.config")
+CODEQL_CLI_PLATFORM_PATH="$BUILD_PATH/tools/codeql-cli/linux"
+CODEQL_CLI_PATH="$CODEQL_CLI_PLATFORM_PATH/$CODEQL_CLI_VERSION"
+CODEQL_CLI_URL=https://github.com/github/codeql-cli-binaries/releases/download/$CODEQL_CLI_VERSION/codeql-linux64.zip
+CODEQL_LIBRARY_PATH="$BUILD_PATH/codeql"
+
+if [ ! -d "$CODEQL_LIBRARY_PATH" ]; then
+    git clone https://github.com/github/codeql.git "$CODEQL_LIBRARY_PATH"
```

Comment:
Please do not script cloning a repo under the openenclave repo; there are more standard mechanisms for this. @BRMcLaren depending on how you think this can best be integrated with CI/CD, I can see two ways of more cleanly maintaining this set of changes:

1. If the CodeQL is expected to run on every build or PR merge and be incorporated into a maintainer's inner loop:

   - Add the codeql project as an explicit git submodule to the openenenclave project. See `tools/oeedger8r-cpp` as an example.  The build and use of the tool is then output to the build folder (including the .cache, _lgtm_build_dir, and _lgtm_detected_source_root currently still being generated in the openenclave source tree) as part of the CMake build hierarchy. 
   - The tests/security block doesn't seem to build by default (not included in tests/CMakeLists.txt), and based on the documentation, it seems like this is intentionally only for manual execution. If that's the case, I would suggest:

2. If CodeQL is expected to be manually set up and run, or triggered as a special workflow in CI/CD:

   - Create an openenclave-codeql project as a sibling project to the openenclave project under the openenclave org (e.g. https://github.com/openenclave/openenclave-codeql)
      - Even in this case, I would suggest making the codeql a submodule of the openenclave-codeql project instead of forking the codeql repo directly because it should keep all of the additional configurations and rules (i.e. changes in this PR) separate from the actual codeql sources.
      - This project can have its own setup/build scripts, and can also reference the openenclave project itself as a submodule to run the static analysis over. 
      - For CI/CD, this potentially decouples the CodeQL runs from the large configuration matrix of OS/Version/Compilation options; in general, the static analysis can be encapsulated by the project to only run over whatever compilation variants configured in the openenclave-codeql project, and CI/CD can simply target and build that.
   - See https://github.com/openenclave/oeedger8r-cpp for an example.

3. Note that the two options need not be mutually exclusive, though either alone should suffice. Again, oeedger8r-cpp shows the integration of both.

Reply:
I second  @CodeMonkeyLeet's suggestions above.